### PR TITLE
server: better UX when remote ID already exists

### PR DESCRIPTION
**Pull request by bobbyrullo**
Instead of cryptic message with nowhere to, give them the choice to
login with that account or register.

### CODE REVIEW

**Comment by bobbyrullo** [comment]
fixes #166 ?

**Comment by bobbyrullo** [comment]
Ok, now you get taken to a page where you can login directly. Not sure how I feel about it though - a new session has to be created, and you get "logged in" in dex's mind even if you don't click the link. Take a look and tell me what you think.

File: server/register.go [hunk 1]
```diff
@@ -92,6 +98,42 @@ func handleRegisterFunc(s *Server) http.HandlerFunc {
...
+				internalError(w, err)
+				return
+			}
+			// make sure to clean up the old session
+			if err = s.KillSession(code); err != nil {
+				internalError(w, err)
+			}
+
+			// finally, we can create a valid redirect URL for them.
+			redirURL, err := s.Login(ses.Identity, newSessionKey)
```
Comment (sym3tri) [inline server/register.go]: forget to handle this error?

File: server/register.go [hunk 2]
```diff
@@ -304,3 +351,20 @@ func newLoginURLFromSession(issuer url.URL, ses *session.Session, register bool,
 	loginURL.RawQuery = v.Encode()
 	return &loginURL
 }
+
+func remoteIdentityExists(ur user.UserRepo, connectorID, id string) (error, bool) {
```
Comment (sym3tri) [inline server/register.go]: Wouldn't it be more idiomatic to have the return values be `(bool, error)`?

**Comment by sym3tri** [comment]
I dig it.

File: server/register.go (hunk 2 above)
Comment (bobbyrullo) [inline server/register.go]: hah yes. Clearly haven't been writing enough code lately

**Pull request merged**